src/subscriber.ts

```diff
@@ -45,23 +52,120 @@ export class Subscriber {
 
     private async _initInstanceVariables(): Promise<void>{
       this.sessionIndex = await this.api.query.session.currentIndex();
-      this.logger.info(
+      this.logger.debug(
         `Session index: ${this.sessionIndex}`
       );
     }
 
+    private async _initPersistence(): Promise<void> {
+      this._initPersistenceDir();
+      await storage.init( {dir:this.PERSISTENCE_DIR} );
+    }
+
+    private _initPersistenceDir(): void{
+      if (!fs.existsSync(this.PERSISTENCE_DIR)) {
+        fs.mkdirSync(this.PERSISTENCE_DIR)
+      }
+    }
+
+    private async _triggerDebugActions(): Promise<void>{
+      this.logger.debug('debug mode active')
+
+      let identity = await this._getIdentity('DVZTk1e5J42EW7LWsAFjyGa33KCC4RjfhU3ux8W1C1FYw2f')
+      this.logger.debug('identity:' +identity)
+
+      identity = await this._getIdentity('CoqkA7rw8zp3mzNfRFnrH5z6uuWyxC4YsZRGtrC73RRwYCG')
+      this.logger.debug(identity.isEmpty ? 'true' : 'false')
+
+    }
+
     private async _handleNewHeadSubscriptions(): Promise<void> {
 
       this.api.rpc.chain.subscribeNewHeads(async (header) => {
-        this.logger.info(
+        this.logger.debug(
           `New header received: ${header}`
         );
-        //Here there will be the handler
+
+        this._performNewChallengeAttempts()
       })
     }
 
-    private async _triggerDebugActions(): Promise<void>{
-      this.logger.debug('debug mode active')
+    private async _handleEventsSubscriptions(): Promise<void> {
+      this.api.query.system.events((events) => {
+
+        events.forEach(async (record) => {
+          const { event } = record;
+          
+          if (this._isJudgementRequestedEvent(event)) {
+            await this._judgementRequestedHandler(event)
+          }
+        })
+      })
+    }
+
+    private _isJudgementRequestedEvent(event: Event): boolean {
+      return event.section == 'identity' && event.method == 'JudgementRequested';
+    }
+
+    private async _judgementRequestedHandler(event: Event): Promise<void>{
+      this.logger.info('New JudgementRequested')
+      const request = this._extractJudgementRequest(event)
+      this._saveJudgementRequest(request)
+      
+    }
+    
+    private _extractJudgementRequest(event: Event): JudgementRequest{
+      const accountId = event.data[0].toString()
+      const registrarIndex = event.data[1].toString()
```

Comment:
What happens if the request is not meant for us, shouldn't we have our registrar index stored and check against it?